src/legendre_fet.cpp

```diff
@@ -0,0 +1,271 @@
+#include <tallies/itally.hpp>
+#include <tallies/legendre_fet.hpp>
+#include <utils/legendre.hpp>
+
+#include <boost/container/static_vector.hpp>
+using StaticVector6 = boost::container::static_vector<size_t, 6>;
+
+LegendreFET::LegendreFET(std::shared_ptr<CartesianFilter> position_filter_,
+                         std::shared_ptr<EnergyFilter> energy_in,
+                         std::vector<LegendreFET::Axis> axes_,
+                         size_t FET_order_, LegendreFET::Quantity quantity_,
+                         LegendreFET::Estimator estimator_, std::string name_)
+    : ITally(quantity_, estimator_, name_),
+      cartesian_filter_(position_filter_),
+      energy_in_(energy_in),
+      axes(),
+      FET_order(FET_order_) {
+  // get the shape or dimensions for the cartesian_filter_
+  // the Legendre-FET requires the co-ordinate information
+  StaticVector3 position_shape;
+  if (cartesian_filter_) {
+    position_shape = position_filter_->get_shape();
+  } else {
+    fatal_error("For the tally, " + name_ +
+                ", the positional filter is required to do the legendre "
+                "functional expansion tally.");
+  }
+
+  StaticVector6 tally_dimensions_(position_shape.begin(), position_shape.end());
+
+  // add the dimension for energy_in_ only if exist
+  if (energy_in_) {
+    std::size_t ne = energy_in_->size();
+    tally_dimensions_.insert(tally_dimensions_.begin(), ne);
+  }
+
+  // for legendre, check the size of axis vector should be between than 3.
+  for (auto& c : axes_) {
+    axes.push_back(c);
+  }
+  if (axes.size() <= 1 && axes.size() >= 3)
+    fatal_error(
+        "The no. of given axis for legendre-FET is not between 1 to 3 for "
+        "tally " +
+        name_ + ".");
+
+  // Add the dimension for axis only when greater than 1
+  std::size_t n_axis = axes.size();
+  if (n_axis == 0) {
+    fatal_error("For tally, " + name_ + " length of axis is zero.");
+  }
+  tally_dimensions_.push_back(n_axis);
+
+  // Add the dimension for the order of FET only if greater than 0
+  std::size_t fet_order_dimension = FET_order_ + 1;
+  tally_dimensions_.push_back(fet_order_dimension);
+
+  // reallocate and fill with zeross for the tally avg, gen-score and variance
+  tally_avg.reallocate(tally_dimensions_);
+  tally_avg.fill(0.0);
+
+  tally_gen_score.reallocate(tally_dimensions_);
+  tally_gen_score.fill(0.0);
+
+  tally_var.reallocate(tally_dimensions_);
+  tally_var.fill(0.0);
+}
+
+void LegendreFET::score_collision(const Particle& p, const Tracker& tktr,
+                                  MaterialHelper& mat) {
+  if (!(estimator_ == LegendreFET::Estimator::Collision)) {
```

Comment:
```suggestion
  if (estimator_ != Estimator::Collision) {
```
I don't think you need the `LegendreFET` namespace there.